spec/controllers/embed_controller_spec.rb

```diff
@@ -20,11 +20,15 @@
 
     before do
       Fabricate(:embeddable_host)
+      SiteSetting.stubs(:embed_link_to_top).returns(true)
       controller.request.stubs(:referer).returns('http://eviltrout.com/some-page')
     end
 
     it "allows a topic to be embedded by id" do
-      topic = Fabricate(:topic)
+      topic = Fabricate(:topic) do
```

Comment:
This could probably be done as `post = Fabricate(:post)` which automatically receives a `topic`. Then if you need the topic do `post.topic`